src/main/java/fi/aalto/cs/apluscourses/ui/APlusAuthenticationView.java

```diff
@@ -53,10 +55,16 @@ protected JComponent createCenterPanel() {
   @Nullable
   @Override
   protected ValidationInfo doValidate() {
-    if (inputField.getPassword().length != 0) {
-      return null;
+    int length = inputField.getPassword().length;
+    int maxLength = authenticationViewModel.getMaxLength();
+    if (length == 0) {
+      return new ValidationInfo("Token must not be empty", inputField);
     }
-    return new ValidationInfo("Token must not be empty", inputField);
+    if (length > maxLength) {
+      return new ValidationInfo(String.format("Token must not be longer than %d characters",
+          maxLength), inputField);
```

Comment:
Nit: `else if`?

Reply:
I like it more without `else`. Or if we had `else if`, I'd put the final `return null` statement also in an `else` branch. Or what about
```java
return length == 0 ? new ValidationInfo(...)
  : length > maxLength ? new ValidationInfo(...)
  : null;
```

Reply:
Well, compared to the ternaries, rather keep it as is :smile:

Reply:
I think mutual exclusiveness of conditions doesn't matter here because each `if` branch ends with `return` statement. I'm pretty sure, in this case, all three options (without `else`, with `else`, and `? :`) compile to the exactly same bytecode, so the question is only about clarity of code.

Reply:
I was also referring to the clarify of the code, not actual bytecode, but I also completely forgot that the first if-branch has a `return` in it, which renders my point moot, my bad